docs: add `x/reports` specs (#957)
desmos-labs/desmos: x/reports/spec/02-concepts.md

---
id: concepts
title: Concepts
sidebar_label: Concepts
slug: concepts
---

# Concepts 

## Report

> A Report is the structure representing a user's report towards a post or another user. The report contains the
> necessary information to identify the subject and reason that lead to its creation.

Reviewer: ```suggestion
A report represents the way that users have to report misbehaving users or bad contents inside Desmos. 
Each report contains the necessary information to identify the target and reason that lead to its creation.
```

### Subspace ID

> The [subspace] ID indicates the ID of the Dapp where the report has been made.

Reviewer: ```suggestion
The subspace id represents the id of the subspace where the report has been created.
```

### ID

> The unique ID that identifies the report itself. This ID is automatically assigned to the report at the moment of its
> creation in a sequential way (e.g. if there's 2 reports to a post, the one we are creating will have id equal to 3).

Reviewer: ```suggestion
The most important thing about a report is its id. This is a unique identifier across the subspace that is used to uniquely reference the report itself along with the subspace id. Report ids are assigned automatically during the handling of a `MsgCreateReport`. 
```

### Reasons IDs

> The IDs array of the reasons this report has been created for. Each ID resolve to a specific reason registered within the
> module.

Reviewer: ```suggestion
The reasons ids represent the array of the reasons that this report has been created for. Each id references a specific reason registered within the subspace of this report.
```

### Message (Optional)

> An optional message to further describe the report's reason.

Reviewer: ```suggestion
A report message can be optionally used to further describe why a user or content has been reported. This can be useful if the reporter wants to leave a message to other users or administrators that will deal with the report itself.
```

### Reporter

> The address of the user that made the report.

Reviewer: ```suggestion
The address of the user that has created the report.
```

### Target

> The target user or post that has been reported.

Reviewer: ```suggestion
A report target represents the content that has been reported. This can be either a [UserTarget](#UserTarget) or a [PostTarget](#PostTarget).
```

### Creation Date

> The creation date of the report.

Reviewer: ```suggestion
The creation date of a report represents the block time at which the report has been stored on the chain. This cannot be specified externally and is assigned automatically when handling a `MsgCreateReport`.
```

## User Target

> The user target contains the data of the reported user.

Reviewer: ```suggestion
A user target object should be used when reporting a specific user.
```

### User
The address of the reported user.

## Post Target

> The post target contains the data of the reported post.

Reviewer: ```suggestion
A post target should be used when reporting a specific post within the same subspace where the report has been created.
```

### Post ID
The ID of the reported post.

## Reason

> A Reason is the structure representing the motivation behind a report.

Reviewer: ```suggestion
A reason is the structure representing the motivation behind a report.
```

### Subspace ID

> The [subspace] ID indicates the ID of the Dapp where the reason lives.

Reviewer: ```suggestion
The subspace id of a reason represents the subspace inside which this reason is valid. Since subspaces can have very different Term of Services from one another, each of them should register their reasons independently so that users are limited in why a report can be created. 
```

### ID

> The unique ID that identifies the reason itself. This ID is automatically assigned to the reason at the moment of its
> creation in a sequential way, just like the report ID.

Reviewer: ```suggestion
A reason id represents the unique id within the subspace that can be used to uniquely reference the registered reason. This is assigned automatically when handling either a `MsgSupportStandardReason` or `MsgAddReason` message.
```

### Title

> The title of the reason.

Reviewer: ```suggestion
The title of a reason should be used to give users a quick idea about why they might want to select this reason during the report creation process. Good titles should be short and easy to understand (e.g. __Spam__, __Explicit content__, etc). 
```

### Description (Optional)

> AN optional extended description of the reason and why the report has been made.

Reviewer: ```suggestion
A reason description can be optionally used to allow the users to further understand when they should select this reason during the report creation process.
```
